### DISCUSSION TITLE
Any good idea on debugging?

### DISCUSSION THREAD

**Discussion by rongcuid** [lwjglgamedev/vulkanbook#46] [Q&A]
I have followed Chapter 7 to an end (which is a looooongg chapter). Unfortunately, when I finally hit that run button... the model doesn't show. There's only the clear color, not the cube:

![image](https://user-images.githubusercontent.com/1312525/209438686-c09a9cb7-0c23-48ea-a87e-90e3a8a3766a.png)

I lost clue on how to debug this issue. There are no freezes, errors, etc, so I don't even know where to start. I _think_ all the commands are recorded correctly (after some stepping) for the cube, but it is just not on screen.

You may look at my code if you like, but note it's in Kotlin: https://github.com/rongcuid/lwjgl-vulkan/tree/c6832c1ee07750ff1253bed0f16b39fe0aa50206

**Comment by rongcuid** [comment]
I fixed a place where I registered texture coordinates as `R32G32B32`, but still doesn't show anything

**Comment by rongcuid** [comment]
Oh, I found it. I set the position to z=2 instead of z=-2. With both problems fixed I see the cube now.

**Comment by lwjglgamedev** [comment]
You should try Renderdoc. It will help you debug the issue. I'm trying to use it but I'm not able to use it with gradle. I'm trying with maven to chekc if  I can see the error.
